lib/core/view_models/login_view_model.dart

```diff
@@ -1,54 +1,54 @@
-import 'package:flutter/material.dart';
 import 'package:pedantic/pedantic.dart';
 import 'package:provider_start/core/constant/view_routes.dart';
 import 'package:provider_start/core/enums/view_state.dart';
 import 'package:provider_start/core/exceptions/auth_exception.dart';
 import 'package:provider_start/core/mixins/validators.dart';
 import 'package:provider_start/core/services/auth/auth_service.dart';
 import 'package:provider_start/core/services/navigation/navigation_service.dart';
+import 'package:provider_start/core/utils/simple_log_printer.dart';
 import 'package:provider_start/core/view_models/base_view_model.dart';
 import 'package:provider_start/locator.dart';
 
 class LoginViewModel extends BaseViewModel with Validators {
-  final _navigationService = locator<NavigationService>();
+  final _log = getLogger();
   final _authService = locator<AuthService>();
+  final _navigationService = locator<NavigationService>();
 
-  final _emailController = TextEditingController();
-  final _passwordController = TextEditingController();
-  final _passwordFocusNode = FocusNode();
-  final _formKey = GlobalKey<FormState>();
-  final _viewFocusNode = FocusNode();
+  String _email = '';
+  String get email => _email;
 
-  TextEditingController get emailController => _emailController;
-  TextEditingController get passwordController => _passwordController;
-  FocusNode get passwordFocusNode => _passwordFocusNode;
-  GlobalKey<FormState> get formKey => _formKey;
-  FocusNode get viewFocusNode => _viewFocusNode;
+  String _password = '';
+  String get password => _password;
 
-  Future<void> login() async {
-    if (!_formKey.currentState.validate()) return;
+  bool _canSubmitEmail = false;
+  bool _canSubmitPassword = false;
+  bool get canLoginUser => _canSubmitEmail && _canSubmitPassword;
 
-    setState(ViewState.Busy);
+  void setEmail(String email) {
+    _log.d('email:$email');
+    _email = email;
+    _canSubmitEmail = validateEmail(email) == null;
+    notifyListeners();
+  }
+
+  void setPassword(String password) {
+    _log.d('password:$password');
+    _password = password;
+    _canSubmitPassword = validatePassword(password) == null;
+    notifyListeners();
+  }
 
-    final email = _emailController.text.trim();
-    final password = _passwordController.text.trim();
+  Future<void> login() async {
+    if (!canLoginUser) return;
+    
+    _log.d('email:$email password:$password');
+    setState(ViewState.Busy);
 
     try {
       await _authService.signInWithEmailAndPassword(email, password);
-      setState(ViewState.Idle);
-
       unawaited(_navigationService.popAllAndPushNamed(ViewRoutes.splash));
     } on AuthException {
       setState(ViewState.Error);
     }
   }
-
-  @override
-  void dispose() {
-    _viewFocusNode.dispose();
-    _emailController.dispose();
-    _passwordController.dispose();
-    _passwordFocusNode.dispose();
-    super.dispose();
-  }
```

Comment:
This controllers still need to disposed, so the login view will have to become stateful.

Reply:
Maybe, do you have a way to confirm that the controllers are not disposing when the view is disposed?